contracts/src/LightClient.sol

```diff
@@ -341,4 +367,77 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
             revert NoChangeRequired();
         }
     }
+
+    /// @notice updates the delay threshold which determines if hotshot is down
+    /// @dev if the number of blocks between the last l1 update and the number of blocks that has
+    /// passed since is > the delayThreshold then hotshot is considered down
+    function updateDelayThreshold(uint32 newThreshold) public onlyOwner {
+        if (newThreshold == delayThreshold) {
+            revert NoChangeRequired();
+        }
+        if (newThreshold < 1 || newThreshold > type(uint32).max) {
+            revert InvalidDelayThreshold();
+        }
+        delayThreshold = newThreshold;
+        emit NewDelayThreshold(newThreshold);
+    }
+
+    /// @notice checks the L1BlockUpdates array and the delayThreshold to determine if hotshot was
+    /// down at a specified L1 block number
+    function isHotshotAlive(uint256 l1BlockNumber) public view returns (bool) {
+        // The block being requested is in the future so return false
+        // if(l1BlockNumber>block.number){
+        //     return true;
+        // }
+
+        // If there is only one block recorded then use the block ledger's current block as the
+        // 'nextBlock'
+        if (l1BlockUpdates.length == 1) {
+            if (
+                block.number - l1BlockUpdates[0] >= delayThreshold
+                    && l1BlockNumber - l1BlockUpdates[0] >= delayThreshold
+            ) {
+                return false;
+            }
+            return true;
+        }
+
+        uint256 prevBlock;
+        uint256 nextBlock;
+
+        /**
+         * TODO: a check for a block number that's too small? e.g before the genesis block was
+         * recorded?
+         */
+        for (uint256 i = 0; i < (l1BlockUpdates.length - 1); i++) {
+            prevBlock = l1BlockUpdates[i];
+            nextBlock = l1BlockUpdates[i + 1];
+
+            //if we're at the last item in the array, the nextBlock is the block ledger's block
+            // number
+            if (i == l1BlockUpdates.length - 1) {
+                nextBlock = block.number;
+            }
+
+            if (prevBlock <= l1BlockNumber && nextBlock >= l1BlockNumber) {
```

Comment:
I think you can condense the first [check](https://github.com/EspressoSystems/espresso-sequencer/pull/1493/files#diff-d85261c27bf48fe0ec00e8cc9bb3ca101d708ec68c85968531b2e2cf4f9ed292R395) with this loop in the following way:
1. Scan from end of the block updates array until you find the first block less than l1BlockNumber at index i. (edge case: no such block is found, this means that you are querying for a block too early in history, throw an error). 
2. prevBlock = l1BlockUpdates[i]
3. nextBlock = l1BlockUpdates[i+1] OR block.number if i+1 is out of bounds. 
4. If `l1BlockNumber - prevBlock >= delayThreshold`, return false

Reply:
Actually, we don't even need the nextBlock, since all we really need is to calculate the distance between l1BlockNumber and prevBlock, so this should work:
``` solidity
function isHotshotAlive(uint256 l1BlockNumber) public view returns (bool) {
        uint256 prevBlock;
        bool prevBlockFound;

         for (uint256 i = l1BlockUpdates.length - 1; !prevBlockFound;) {
            if (i == 0) {
                break;
            }
            else if (l1BlockUpdates[i] <= l1BlockNumber) {
                prevBlockFound = true;
                prevBlock = l1BlockUpdates[i];
                break;
            }
        }

        // If no snapshot is found, we don't have enough history stored to tell whether HotShot was down.
        if (!prevBlockFound) {
            revert InsufficientSnapshotHistory();
        }

        return l1BlockNumber - prevBlock <= delayThreshold;
    }
```

Reply:
This is a much cleaner way of thinking about it and implementing it! Thanks @nomaxg